bin/zmonlyofficeconfig

```diff
@@ -0,0 +1,256 @@
+  eval "$cmd"
+  RC=$?
+  if [ $RC -ne "0" ]; then
+    STATUS=$R
+    echo "Updating zimbraDocumentEditingJwtSecret faced problem."
+  fi
+
+  #update the default.json
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.browser.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.inbox.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.outbox.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.session.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+}
+
+set_jwt_secret()
+{
+  if [ -z "$1" ]; then
+    echo "Setting Random Secret..."
+    jwtsecret=$(generate_random)
+    update_jwt_config $jwtsecret
+  else
+    jwtsecret=$1
+    update_jwt_config $jwtsecret
+  fi
+}
+
+generate_fonts_data()
+{
+  mkdir -p /opt/zimbra/onlyoffice/documentserver/fonts
+  
+  fonts_gen=$(LD_LIBRARY_PATH=/opt/zimbra/onlyoffice/documentserver/server/FileConverter/bin /opt/zimbra/onlyoffice/documentserver/server/tools/allfontsgen \
+  --input="/opt/zimbra/onlyoffice/documentserver/core-fonts" \
+  --allfonts-web="/opt/zimbra/onlyoffice/documentserver/sdkjs/common/AllFonts.js" \
+  --allfonts="/opt/zimbra/onlyoffice/documentserver/server/FileConverter/bin/AllFonts.js" \
+  --images="/opt/zimbra/onlyoffice/documentserver/sdkjs/common/Images" \
+  --selection="/opt/zimbra/onlyoffice/documentserver/server/FileConverter/bin/font_selection.bin" \
+  --output-web="/opt/zimbra/onlyoffice/documentserver/fonts" \
+  --use-system="true")
+  R=$?
+  if [ $R -ne "0" ]; then
+    STATUS=$R
+    echo "Error while generating fonts. ${fonts_gen}"
+  fi
+
+}
+  
+generate_presentation_themes()
+{
+  presentation_themes_gen=$(LD_LIBRARY_PATH=/opt/zimbra/onlyoffice/documentserver/server/FileConverter/bin /opt/zimbra/onlyoffice/documentserver/server/tools/allthemesgen \
+    --converter-dir="/opt/zimbra/onlyoffice/documentserver/server/FileConverter/bin"\
+    --src="/opt/zimbra/onlyoffice/documentserver/sdkjs/slide/themes"\
+    --output="/opt/zimbra/onlyoffice/documentserver/sdkjs/common/Images")
+  R=$?
+  if [ $R -ne "0" ]; then
+    STATUS=$R
+    echo "Error while generating presentation themes. ${presentation_themes_gen}"
+  fi
+}
+
+type="$1"
+if [ "$#" -eq 0 ]; then
+    type="new"
+fi
+
+
+case "$type" in 
+  new)
+
+    jwtsecret=$(su - zimbra -c "zmprov -m -l getConfig zimbraDocumentEditingJwtSecret")
+    if [ -z "$jwtsecret" ]; then
+      echo "JWT Secret not set."
+      set_jwt_secret
+    else
+      IFS=':'
+      read -ra arr <<< "$jwtsecret"
+      document_editing_JWT_secret=$(echo "${arr[1]}" | xargs)
+      update_jwt_config $document_editing_JWT_secret
```

Comment:
Throughout the code I would recommend to use double quotes for variables..

Reply:
"$document_editing_JWT_secret"

Reply:
I think the standard way for accessing variables is by not using "", but by simply using $

Reply:
Check this https://github.com/ZimbraOS/zm-onlyoffice/pull/2#issuecomment-878998536

Reply:
done